src/nvim/api/private/helpers.c

```diff
@@ -1387,6 +1387,11 @@ void add_user_command(String name, Object command, Dict(user_command) *opts, int
   LuaRef luaref = LUA_NOREF;
   LuaRef compl_luaref = LUA_NOREF;
 
+  if (mb_islower(name.data[0])) {
```

Comment:
hot take: should we allow non-ascii lowercase? I think only the ASCII lowercase range needs to be reserved for the present logic in ex_docmd to work. could be a separate issue/pr of c if this is too controversial.

Reply:
No opinion. Have you got any use cases in mind?